sympy/ntheory/residue_ntheory.py

```diff
@@ -1533,6 +1532,8 @@ def polynomial_congruence(expr, m):
     if rank == 2:
         return quadratic_congruence(0, coefficients[0], coefficients[1],
             m)
+    if coefficients[0] == 1 and all(v == 0 for v in coefficients[1:-1]):
```

Comment:
```suggestion
    if coefficients[0] == 1 and 1 + coefficients[-1] = sum(coefficients):
```

Reply:
When the coefficient is negative this might give wrong result. I think I should take mod m of all coefficients then this suggestion can be correctly implemented.

Reply:
No, what you have is better.

Reply:
But why not
```
if not any(coefficients[1: -1]):
    g = igcd(*coefficients)
    if coefficients[0]//g == 1:
        return nthroot_mod(-coefficients[-1]//g, rank - 1, m, True)
```
In that case `2*x**2 + 8` will be reduced to `x**2 + 4`.

Reply:
yes but in this case also we have to take `mod m` of the coefficients. For example:
for `x**5 + 16 x + 7 = 0 mod 8`. But this can be reduced to `x**5 + 7`. But (`if not any(coefficients[1:-1])`) will not catch this unless `coefficients = coefficients mod m`.
Also by dividing the `coefficients[0] with gcd` might result in loss of some roots.
For example : `2*x**2 + 8 = 0 mod 10`has roots [1,4,6,9] but `x**2 + 4 = 0 mod 10 ` has roots
roots [4,6]  because `mod_inv(2,10)` does not exist.